core/templates/components/rubrics-editor/rubrics-editor.directive.ts

```diff
@@ -65,40 +65,86 @@ angular.module('oppia').directive('rubricsEditor', [
 
-          ctrl.cancelEditExplanation = function(index) {
-            ctrl.editableExplanation[index] = explanationMemento[index];
-            ctrl.explanationEditorIsOpen[index] = false;
+          ctrl.cancelEditExplanation = function(difficulty, index) {
+            ctrl.editableExplanations[difficulty][index] = (
+              explanationsMemento[difficulty][index]);
+            ctrl.explanationEditorIsOpen[difficulty][index] = false;
+          };
+
+          ctrl.addExplanationForDifficulty = function(difficulty) {
+            ctrl.editableExplanations[difficulty].push('');
+            ctrl.onSaveRubric(
+              difficulty, ctrl.editableExplanations[difficulty]);
+            explanationsMemento[difficulty] = angular.copy(
+              ctrl.editableExplanations[difficulty]);
+            ctrl.explanationEditorIsOpen[
+              difficulty][
+              ctrl.editableExplanations[difficulty].length - 1] = true;
+          };
+
+          ctrl.deleteExplanation = function(difficulty, index) {
+            ctrl.explanationEditorIsOpen[difficulty][index] = false;
+            ctrl.editableExplanations[difficulty].splice(index, 1);
+            ctrl.onSaveRubric(
+              difficulty, ctrl.editableExplanations[difficulty]);
+            explanationsMemento[difficulty] = angular.copy(
+              ctrl.editableExplanations[difficulty]);
           };
+
+          ctrl.isAnyExplanationEmptyForDifficulty = function(difficulty) {
+            for (var idx in explanationsMemento[difficulty]) {
+              if (
+                ctrl.isExplanationEmpty(
+                  explanationsMemento[difficulty][idx])) {
+                return true;
+              }
+            }
+            return false;
+          };
+
           ctrl.$onInit = function() {
-            ctrl.explanationEditorIsOpen = [false, false, false];
-            ctrl.editableExplanation = [null, null, null];
-            ctrl.activeRubricIndex = 0;
+            ctrl.explanationEditorIsOpen = {};
+            ctrl.editableExplanations = {};
+            for (var idx in ctrl.getRubrics()) {
+              var explanations = ctrl.getRubrics()[idx].getExplanations();
+              var difficulty = ctrl.getRubrics()[idx].getDifficulty();
+              explanationsMemento[difficulty] = angular.copy(explanations);
+              ctrl.explanationEditorIsOpen[difficulty] = (
+                Array(explanations.length).fill(false));
+              ctrl.editableExplanations[difficulty] = (
+                angular.copy(explanations));
+            }
             ctrl.EXPLANATION_FORM_SCHEMA = {
               type: 'html',
               ui_config: {}
             };
           };
+
+          $scope.$on('skillDescriptionChanged', function(evt) {
+            var explanations = ctrl.getRubrics()[1].getExplanations();
+            var difficulty = ctrl.getRubrics()[1].getDifficulty();
+            explanationsMemento[difficulty] = angular.copy(explanations);
```

Comment:
This should only be updated when the explanations array for "medium" hasn't been filled with anything yet, or if the array is still at its default value, right? (Otherwise a typo fix to the skill might cause new content that the creator has added since then to be overwritten.)

Reply:
Done, if the first 'medium' explanation is edited or deleted, autofill is stopped.